plugins/SegmentEditor/javascripts/Segmentation.js

```diff
@@ -211,21 +204,34 @@ Segmentation = (function($) {
                     }
 
 
-                    injClass = "";
+                    injClass = [];
                     var checkSelected = segment.definition;
+                    var escapedSegmentName = (segment.definition).replace(/"/g, '&quot;');
 
-                    if( checkSelected == self.currentSegmentStr ||
-                        checkSelected == decodeURIComponent(self.currentSegmentStr)
-                    ) {
-                        injClass = 'class="segmentSelected"';
+                    if (checkSelected === self.currentSegmentStr || checkSelected === decodeURIComponent(self.currentSegmentStr)) {
+                        injClass.push('segmentSelected');
+                    }
+                    if (segment.starred) {
+                        injClass.push('segmentStarred');
                     }
-                    listHtml += '<li data-idsegment="'+segment.idsegment+'" data-definition="'+ (segment.definition).replace(/"/g, '&quot;') +'" '
-                        +injClass+' title="'+ getSegmentTooltipEnrichedWithUsername(segment) +'"><span class="segname" tabindex="4">'+getSegmentName(segment)+'</span>';
-                    if(self.segmentAccess == "write") {
-                        listHtml += '<span class="editSegment" title="'+ self.translations['General_Edit'].toLocaleLowerCase() +'"></span>';
+                    listHtml += '' +
+                      '<li class="' + injClass.join(' ') + '"' +
+                        'data-idsegment="' + segment.idsegment + '"' +
+                        'data-definition="' + escapedSegmentName + '"' +
+                      '>' +
+                        '<span class="segname" tabindex="4" title="' + getSegmentTooltipEnrichedWithUsername(segment) + '" >' + getSegmentName(segment) + '</span>';
+                    if (self.segmentAccess === "write") {
+                        listHtml += '' +
+                          '<button data-star class="starSegment" title="' + self.translations[segment.starred ? 'General_RemoveFromFavorites' : 'General_AddToFavorites'].toLocaleLowerCase() + '">️' +
+                            '<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24">' +
```

Comment:
We should use an existing star icon from the UI icons, or add a new one to it if the existing one can't be used.

`<span class="icon-star"></span>` should work here just fine.

Reply:
In both cases we can not control icons in CSS if it’s cloned via `<use>` or set as a background.
I completed the description with the micro-animations I wanted to add on this icons.
It’s a single path SVG, there should be no problem of having it hundreds of time in the page.